src/sessions.ts

```diff
@@ -492,8 +506,19 @@ export class ClientSession
       bypassPinningCheck: true
     });
 
+    const timeoutMS =
+      typeof options?.timeoutMS === 'number'
+        ? options.timeoutMS
+        : typeof this.timeoutMS === 'number'
+        ? this.timeoutMS
+        : null;
+
+    const timeoutContext = this.timeoutContext?.csotEnabled()
```

Comment:
The only place this.timeoutContext is assigned is in withTransaction.  That means that unless we're in a convenient transaction, `this.timeoutContext?.csotEnabled()` will be falsy.  Even when we're not in a convenient transaciton, we still need to apply a timeout here, right?  for example, the following should apply defaultTimeoutMS to the commit:

```typescript
const session = client.startSession({ defaultTimeoutMS: 1000 });
...
await session.commitTransaction();
```